check_stability.py

```diff
@@ -483,31 +482,69 @@ class LogHandler(reader.LogHandler):
         Subclasses reader.LogHandler.
         """
         def __init__(self):
-            self.results = defaultdict(lambda: defaultdict(lambda: defaultdict(int)))
+            self.results = []
+
+        def find_or_create_test(self, data):
+            for test in self.results:
+                if test["name"] == data["test"]:
+                    return test
+
+            test = {
+                "name": data["test"],
+                "subtests": [],
+                "status": {"OK": 0, "FAIL": 0}
+            }
+            self.results.append(test)
+            return test
+
+        def find_or_create_subtest(self, data):
+            test = self.find_or_create_test(data)
+
+            for subtest in test["subtests"]:
+                if subtest["name"] == data["subtest"]:
+                    return subtest
+
+            subtest = {
+                "name": data["subtest"],
+                "status": {"PASS": 0, "FAIL": 0},
+                "messages": set()
```

Comment:
I was trying to be mindful of the limited space in the GitHub UI, but instead of outputting this:
```
your flakey test failed because of this reason; your flaket test failed because of this other reason
```

we could output something like this without much additional space-usage:

```
your flakey test failed because of this reason (4 time(s)); your flaket test failed because of this other reason (1 time(s))
```